extensions/amp-base-carousel/0.1/test/test-carousel.js

```diff
@@ -236,4 +236,30 @@ describes.realWin('carousel implementation', {}, (env) => {
       expect(carousel.isAtStart()).to.be.false;
     });
   });
+
+  describe('resetScrollReferencePoint_', () => {
+    it('currentElementOffset_ & currentIndex_ should be set when it is a' + 
+    ' programmatic scroll', async () => {
+      setStyle(element, 'width', '299.2px');
```

Comment:
Does this condition happen only when the element width is not floored?

Reply:
No, this is actually unnecessary. I'll simplify.